Add LeetCode-1 (different approach) and LeetCode-167 in C++

r8025n opened the pull request:
resolves #42

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+class Solution {
+public:
+    vector<int> twoSum(vector<int>& nums, int target) {
+        vector<pair<int,int>> valueIndexPair;
```
Change this code to 
```cpp
vector<pair<int, int>> valueIndexPair;
```

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+    vector<int> twoSum(vector<int>& nums, int target) {
+        vector<pair<int,int>> valueIndexPair;
+	    vector<int> returnVector;	
+	    int index1, index2, sum;
```
Fix the indentation.

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+	    int index1, index2, sum;
+
+        for(int k = 0; k < nums.size(); k++) {
+            valueIndexPair.push_back(pair<int,int>(nums[k], k));
```
Change this code to
```cpp
valueIndexPair.push_back(pair<int, int>(nums[k], k));
```

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+        
+        sort(valueIndexPair.begin(), valueIndexPair.end());
+        
+	    index1 = 0;
```
Fix indentation

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+        sort(valueIndexPair.begin(), valueIndexPair.end());
+        
+	    index1 = 0;
+	    index2 = valueIndexPair.size() - 1;
```
Fix indentation

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+        while(index1 < index2) {
+            sum = valueIndexPair[index1].first + valueIndexPair[index2].first;
+            
+	        if(sum == target)
```
Fix indentation

Tahanima reviewed (changes requested):
Hi @r8025n 
Please resolve all the requested changes. Also, change the folder name from `167_TwoSum II` to `167_TwoSumII`

r8025n commented:
HI,
Thank you for your feedback. I have addressed the requested changes, please review.

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+public:
+    vector<int> twoSum(vector<int>& nums, int target) {
+        vector<pair<int, int>> valueIndexPair;
+	vector<int> answer;	
```
Need to fix the indentation.

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+    vector<int> twoSum(vector<int>& nums, int target) {
+        vector<pair<int, int>> valueIndexPair;
+	vector<int> answer;	
+	int index1, index2, sum;
```
Need to fix the indentation.

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+        
+        sort(valueIndexPair.begin(), valueIndexPair.end());
+        
+	index1 = 0;
```
Need to fix the indentation.

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+        sort(valueIndexPair.begin(), valueIndexPair.end());
+        
+	index1 = 0;
+	index2 = valueIndexPair.size() - 1;
```
Need to fix the indentation.

Tahanima commented on Algorithms/Easy/1_TwoSum/SolutionOne.cpp:
```diff
@@ -0,0 +1,33 @@
+            else
+                index2--;
+        }
+        
```
Need to fix the indentation of the if-else block.

Tahanima reviewed (changes requested):
Hi @r8025n 
Please fix all the indentation issues.

r8025n commented:
Hi @Tahanima 
I am kind of lost now, could you please help me out  and point me to the right direction? As your recent requested changes are suggesting there are indentation issues in my code but in my forked repo the code looks okay. What am I missing here?

What I did so far:
1. Opened a pull request. 
2. Some changes were requested by you.
3. Addressed the requested changes in my local git repo, updated my repo with the original upstream repo by fetching and merging, then pushed them to my forked repo.
4. And now as you addressed some indentation issues, as I visited my fork repo they seems fine there!!

So did I miss any steps after pushing the first requested changes to my forked repo?

Tahanima reviewed (approved):
@r8025n I get your point. No worries. I'll fix the indentation after merging. Keep up the good work! 😃

r8025n commented:
Thank You :)
